File: Python/46-孩子们的游戏.py
# -*- coding:utf-8 -*-
class Solution:
    def LastRemaining_Solution(self, n, m):
        # write code here
        if not n or not m:
            return -1
        data = list(range(n))
        idx = 0
        while n > 1:
            idx = (idx + m - 1) % n
            data.pop(idx)
            n -= 1
            idx = idx % n
        return data[0]

File: C++/46-孩子们的游戏.cpp
class Solution
{
public:
    int LastRemaining_Solution(int n, int m)
    {
        if (n == 0 or m == 0)
            return -1;
        vector<int> data;
        for (int i = 0; i < n; i++)
            data.push_back(i);
        int idx = 0;
        while (n > 1)
        {
            idx = (idx + m - 1) % n;
            data.erase(data.begin() + idx);
            n--;
            idx = idx % n;
        }
        return data[0];
    }
};